### ISSUE TITLE
Suggestion for tab behavior

### ISSUE THREAD

**Issue by kennybroh** [mucommander/mucommander#280]
I just started using MUCommander and like it very much.  I am on a Mac but also run a Windows VM under Parallels.  I have been using TotalCommander for many years and the resemblance is quite strong.

As I become more familiar with MUCommander I am sure I will have more suggestions, but I notice one important thing I hope you will consider and that is the behavior of locked tabs.

As you have implemented it now, if a tab is locked and another folder is selected it spawns a new tab, leaving the locked tab in place.  That can lead to an unnecessary proliferation of tabs.   What is a more flexible approach is to have 2 different selections in the right click menu-- "Locked" and "Locked - allow folder change".  The first would work the way it does now, and the second would not spawn a new tab when the folder changed, but clicking on the tab header would return to the locked folder and would always return to that folder on program start.

**Comment by ahadas** [comment]
@kennybroh thanks and please feel free to approach with any enhancement that you think might be useful for the community!

> As you have implemented it now, if a tab is locked and another folder is selected it spawns a new tab, leaving the locked tab in place. That can lead to an unnecessary proliferation of tabs. What is a more flexible approach is to have 2 different selections in the right click menu-- "Locked" and "Locked - allow folder change". The first would work the way it does now, and the second would not spawn a new tab when the folder changed

Let's say that I lock a tab that displays a certain path X. If I don't attempt to enter a new location, there's no difference between the two. However, if I try to enter another location inside this tab - the current locking would open it in a new tab while the suggested locking would no do anything (or generate an error?). What I find hard to see is when would I want the application to prevent me from switching to the new location?

>, but clicking on the tab header would return to the locked folder and would always return to that folder on program start.

Not sure I understand this part. How is it different than the current behavior that by default (unless you set custom folders in the Preferences/Folders dialog) restores the last visited folders, including the locked tab, on startup?

**Comment by ahadas** [comment]
Thanks for suggesting this. It sounds complicated to implement and to use (e.g., how would someone remember what was the original location that will be restored when clicking the header?). I think are easier ways to achieve similar functionality, for instance by setting the location to return to as a bookmark

**Comment by kennybroh** [comment]
I don't think you understand what I'm saying.  Let me show you how it works in Windows Commander:

The normal operation of a locked tab is to display everything in that folder, and if the user selects a subfolder, that appears in a separate tab.  That can lead to a lot of unnecessary tabs.

But if that tab is "Locked" but directory changes are allowed, all navigation stays in that tab.  But the heading of the original tab doesn't change.

***@***.***

[Graphical user interface, text, application  Description automatically generated]

The user doesn't have to remember what the original heading was because it never changes, and clicking on the header restores the location immediately and it is also restored whenever the app restarts.  There's never a need to remember anything or set a bookmark.


                                         So, when I click here:

I
immediately return to the original folder.

[Graphical user interface, text, application  Description automatically generated]

[Graphical user interface, application  Description automatically generated]


Some tabs you would always want stay in their locked position, like the root of a drive or a main folder like "Documents".  But there are other folders a user would want to navigate within one tab.
I don't think this should be terribly hard to implement and it would prevent cluttering the app with multiple tabs.

**Comment by ahadas** [comment]
@kennybroh thanks for the follow up - this aligns well with how I understood it before. I'll elaborate on what I meant - let's say that I locked a tab while being at folder A so the header would just say "A". Then I switch between different folders and the title stays "A" - how would I even remember what "A" was, let's say when I have several folders named "A"? it doesn't sound practical to me - this feature would require to distinguish somehow between regular locked tabs and such weird locked tabs for something that can/should be achieved differently.
That said, if you'd like to POC this feature and it wouldn't introduce too much overhead then it is welcome.

**Comment by kennybroh** [comment]
You would remember what "A" was because the name hasn't changed.

**Comment by kennybroh** [comment]
There is nothing "weird" about a tab that's locked but allows you to navigate without opening new tabs.  That is exactly how Total Commander for Windows works.   MuCommander is so similar in appearance to Total Commander, which has been around for almost 30 years, that I cannot believe you aren't familiar with it.  The English version was first released in 1993 and it is far and away the best file manager of any of them.  Of the Mac products, MuCommander is the closest, but it lacks many important details in how it works and still doesn't have the depth of functionality in Total Commander.

**Comment by ahadas** [comment]
@kennybroh thanks again for the follow up. sure, I haven't used total commander for something like 20 years but I know it :) it's ok to get inspiration from it but we don't try to duplicate total commander.

let's not argue about whether it's weird or not because that's subjective. to me it's weird since I'm used to think about locked tabs as tabs that cannot be modified, like what [the Lock Tab extension to Chrome](https://chrome.google.com/webstore/detail/lock-tab/nhbdiieigbgalknjplfpgmjnpbnkchnb) does. I, for one, wouldn't use the term "lock" for the tabs you describe but rather something like "stateless" - tabs that have some initial state that is restored on startup or when clicking on their headers, but their modification not locked.

I'm re-opening this issue so others could notice it and either pick it up and implement this or get more feedback on that. I, for one, find this confusing but if someone likes to work on it and comes with a reasonable design for this, I'm open to hear
